addons/osfstorage/decorators.py

```diff
@@ -64,14 +81,15 @@ def waterbutler_opt_hook(func):
     @must_be_signed
     @handle_django_errors
     @must_not_be_registration
-    @must_have_addon('osfstorage', 'node')
+    @load_guid_as_target
     @functools.wraps(func)
     def wrapped(payload, *args, **kwargs):
         try:
             user = OSFUser.load(payload['user'])
-            dest_node = AbstractNode.load(payload['destination']['node'])
-            source = OsfStorageFileNode.get(payload['source'], kwargs['node'])
-            dest_parent = OsfStorageFolder.get(payload['destination']['parent'], dest_node)
+
+            dest_target = Guid.load(payload['destination']['target']).referent
+            source = OsfStorageFileNode.get(payload['source'], kwargs['target'])
+            dest_parent = OsfStorageFolder.get(payload['destination']['parent'], dest_target)
```

Comment:
The osfstorage_copy_hook and osfstorage_view_hook were refactored for v2 here #7773. They are not in use yet, so the v1 hooks are still in place. However, the new v2 views adapt the logic from this waterbutler_opt_hook which will no longer be correct. Maybe a follow-up ticket to bring the v2 Move/Copy hooks up-to-date with files on anything?